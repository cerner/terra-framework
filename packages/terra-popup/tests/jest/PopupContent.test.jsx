import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import PopupContent, * as PopupContentExports from '../../src/_PopupContent';

describe('PopupContent', () => {
  const requiredProps = {
    contentHeight: 640,
    contentWidth: 640,
    onRequestClose: jest.fn(),
    onResize: jest.fn(),
  };
  const children = <div>children node(s)</div>;

  describe('Prop Tests', () => {
    describe('with default props', () => {
      const subject = (
        <PopupContent {...requiredProps}>
          {children}
        </PopupContent>
      );
      const wrapper = shallowWithIntl(subject);

      it('matches the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('with header', () => {
      const extraProps = {
        contentHeightMax: 640,
        contentWidthMax: 640,
        refCallback: jest.fn(),
      };
      const subject = (
        <PopupContent {...requiredProps} {...extraProps}>
          {children}
        </PopupContent>
      );

      it('matches the shallow snapshot', () => {
        const wrapper = shallowWithIntl(subject);
        expect(wrapper).toMatchSnapshot();
      });

      it('matches the mount snapshot', () => {
        const wrapper = mountWithIntl(subject);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});

describe('PopupContentExports', () => {
  it('should match the snapshot', () => {
    expect(PopupContentExports).toMatchSnapshot();
  });
});
