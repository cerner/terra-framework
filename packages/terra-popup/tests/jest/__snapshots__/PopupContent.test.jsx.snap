// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupContent Prop Tests with default props matches the snapshot 1`] = `
<FocusTrap
  _createFocusTrap={[Function]}
  active={true}
  focusTrapOptions={
    Object {
      "clickOutsideDeactivates": true,
      "returnFocusOnDeactivate": true,
    }
  }
  paused={false}
>
  <div>
    <OnClickOutside(HookshotContent)
      className="content"
      data-terra-popup-content={true}
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onEsc={[MockFunction]}
      onResize={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      role="dialog"
      stopPropagation={false}
      tabIndex="0"
    >
      <div
        className="inner"
        style={
          Object {
            "height": "640px",
            "width": "640px",
          }
        }
      >
        <div
          key=".0"
        >
          children node(s)
        </div>
      </div>
    </OnClickOutside(HookshotContent)>
  </div>
</FocusTrap>
`;

exports[`PopupContent Prop Tests with header matches the mount snapshot 1`] = `
<PopupContent
  classNameInner={null}
  contentHeight={640}
  contentHeightMax={640}
  contentWidth={640}
  contentWidthMax={640}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isFocusedDisabled={false}
  isHeaderDisabled={false}
  isHeightAutomatic={false}
  isWidthAutomatic={false}
  onRequestClose={[MockFunction]}
  onResize={[MockFunction]}
  refCallback={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="content content"
            data-terra-popup-content="true"
            intl="[object Object]"
            role="dialog"
            tabindex="0"
          >
            <div
              class="inner is-full-screen"
              style="height: 640px; width: 640px;"
            >
              <div
                class="content-container-fill"
              >
                <div
                  class="header"
                >
                  <div
                    class="header"
                  >
                    <button
                      aria-disabled="false"
                      aria-label="Terra.popup.header.close"
                      class="button utility"
                      title="Terra.popup.header.close"
                      type="button"
                    >
                      <span
                        class="button-label icon-only"
                      >
                        <span
                          class="icon"
                        >
                          <span
                            class="close-icon icon-svg"
                          />
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="main"
                >
                  <div
                    class="scroll normalizer"
                  >
                    <div>
                      children node(s)
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div>
      <OnClickOutside(HookshotContent)
        className="content"
        data-terra-popup-content={true}
        eventTypes={
          Array [
            "mousedown",
            "touchstart",
          ]
        }
        excludeScrollbar={false}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": null,
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
        onEsc={[MockFunction]}
        onResize={[Function]}
        outsideClickIgnoreClass="ignore-react-onclickoutside"
        preventDefault={false}
        refCallback={
          [MockFunction] {
            "calls": Array [
              Array [
                <div
                  class="content content"
                  data-terra-popup-content="true"
                  intl="[object Object]"
                  role="dialog"
                  tabindex="0"
                >
                  <div
                    class="inner is-full-screen"
                    style="height: 640px; width: 640px;"
                  >
                    <div
                      class="content-container-fill"
                    >
                      <div
                        class="header"
                      >
                        <div
                          class="header"
                        >
                          <button
                            aria-disabled="false"
                            aria-label="Terra.popup.header.close"
                            class="button utility"
                            title="Terra.popup.header.close"
                            type="button"
                          >
                            <span
                              class="button-label icon-only"
                            >
                              <span
                                class="icon"
                              >
                                <span
                                  class="close-icon icon-svg"
                                />
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                      <div
                        class="main"
                      >
                        <div
                          class="scroll normalizer"
                        >
                          <div>
                            children node(s)
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        role="dialog"
        stopPropagation={false}
        tabIndex="0"
      >
        <HookshotContent
          className="content"
          data-terra-popup-content={true}
          disableOnClickOutside={[Function]}
          enableOnClickOutside={[Function]}
          eventTypes={
            Array [
              "mousedown",
              "touchstart",
            ]
          }
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": null,
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
          onEsc={[MockFunction]}
          onResize={[Function]}
          outsideClickIgnoreClass="ignore-react-onclickoutside"
          preventDefault={false}
          refCallback={
            [MockFunction] {
              "calls": Array [
                Array [
                  <div
                    class="content content"
                    data-terra-popup-content="true"
                    intl="[object Object]"
                    role="dialog"
                    tabindex="0"
                  >
                    <div
                      class="inner is-full-screen"
                      style="height: 640px; width: 640px;"
                    >
                      <div
                        class="content-container-fill"
                      >
                        <div
                          class="header"
                        >
                          <div
                            class="header"
                          >
                            <button
                              aria-disabled="false"
                              aria-label="Terra.popup.header.close"
                              class="button utility"
                              title="Terra.popup.header.close"
                              type="button"
                            >
                              <span
                                class="button-label icon-only"
                              >
                                <span
                                  class="icon"
                                >
                                  <span
                                    class="close-icon icon-svg"
                                  />
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                        <div
                          class="main"
                        >
                          <div
                            class="scroll normalizer"
                          >
                            <div>
                              children node(s)
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>,
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          role="dialog"
          stopPropagation={false}
          tabIndex="0"
        >
          <div
            className="content content"
            data-terra-popup-content={true}
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": null,
                "now": [Function],
                "onError": [Function],
                "textComponent": "span",
                "timeZone": null,
              }
            }
            role="dialog"
            tabIndex="0"
          >
            <div
              className="inner is-full-screen"
              style={
                Object {
                  "height": "640px",
                  "width": "640px",
                }
              }
            >
              <ContentContainer
                fill={true}
                header={
                  <div
                    className="header"
                  >
                    <FormattedMessage
                      id="Terra.popup.header.close"
                      values={Object {}}
                    >
                      [Function]
                    </FormattedMessage>
                  </div>
                }
              >
                <div
                  className="content-container-fill"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="header"
                    >
                      <FormattedMessage
                        id="Terra.popup.header.close"
                        values={Object {}}
                      >
                        <Button
                          icon={
                            <span
                              className="close-icon"
                            />
                          }
                          isBlock={false}
                          isCompact={false}
                          isDisabled={false}
                          isIconOnly={true}
                          isReversed={false}
                          onClick={[MockFunction]}
                          text="Terra.popup.header.close"
                          type="button"
                          variant="utility"
                        >
                          <button
                            aria-disabled={false}
                            aria-label="Terra.popup.header.close"
                            className="button utility"
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[MockFunction]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            title="Terra.popup.header.close"
                            type="button"
                          >
                            <span
                              className="button-label icon-only"
                            >
                              <span
                                className="icon"
                              >
                                <span
                                  className="close-icon icon-svg"
                                />
                              </span>
                            </span>
                          </button>
                        </Button>
                      </FormattedMessage>
                    </div>
                  </div>
                  <div
                    className="main"
                  >
                    <Scroll
                      className="normalizer"
                    >
                      <div
                        className="scroll normalizer"
                      >
                        <div
                          key=".0"
                        >
                          children node(s)
                        </div>
                      </div>
                    </Scroll>
                  </div>
                </div>
              </ContentContainer>
            </div>
          </div>
        </HookshotContent>
      </OnClickOutside(HookshotContent)>
    </div>
  </FocusTrap>
</PopupContent>
`;

exports[`PopupContent Prop Tests with header matches the shallow snapshot 1`] = `
<FocusTrap
  _createFocusTrap={[Function]}
  active={true}
  focusTrapOptions={
    Object {
      "clickOutsideDeactivates": true,
      "returnFocusOnDeactivate": true,
    }
  }
  paused={false}
>
  <div>
    <OnClickOutside(HookshotContent)
      className="content"
      data-terra-popup-content={true}
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onEsc={[MockFunction]}
      onResize={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      refCallback={[MockFunction]}
      role="dialog"
      stopPropagation={false}
      tabIndex="0"
    >
      <div
        className="inner is-full-screen"
        style={
          Object {
            "height": "640px",
            "width": "640px",
          }
        }
      >
        <ContentContainer
          fill={true}
          header={
            <div
              className="header"
            >
              <FormattedMessage
                id="Terra.popup.header.close"
                values={Object {}}
              >
                [Function]
              </FormattedMessage>
            </div>
          }
        >
          <div
            key=".0"
          >
            children node(s)
          </div>
        </ContentContainer>
      </div>
    </OnClickOutside(HookshotContent)>
  </div>
</FocusTrap>
`;

exports[`PopupContentExports should match the snapshot 1`] = `
Object {
  "cornerSize": 3,
  "default": [Function],
}
`;
